.github/workflows/roll_torch_mlir.yml

```diff
@@ -0,0 +1,65 @@
+name: Roll PyTorch
+
+on:
+  schedule:
+    - cron: '0 13 * * *'
+  workflow_dispatch:
+  pull_request:
+
+jobs:
+  roll_torch_mlir:
+    runs-on: ubuntu-20.04-64core
+    # Don't run this in everyone's forks.
+    if: github.repository == 'openxla/iree'
+
+    steps:
+
+      - name: Prepare workspace
+        run: |
+          # Clear the workspace directory so that we don't run into errors about
+          # existing lock files.
+          sudo rm -rf $GITHUB_WORKSPACE/*
+
+      - name: Get iree
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c
+        with:
+          submodules: 'true'
+          ref: main
+          fetch-depth: 0
+
+
+      - name: Create pull request
+        env:
+          GH_TOKEN: ${{ secrets.WRITE_ACCESS_TOKEN }}
+          APPROVAL_TOKEN: ${{ github.token }}
+        run: |
+          git config --local user.email "iree-github-actions-bot@google.com"
+          git config --local user.name "IREE torch-mlir Roller"
+          git submodule update --init --recursive third_party/torch-mlir
+          echo "entering torch-mlir"
+          cd third_party/torch-mlir
+          initial_sha="$(git rev-parse --short HEAD)"
+          echo "$initial_sha"
+          git fetch --tags origin
+          git checkout origin/main
+          new_sha="$(git rev-parse --short HEAD)"
+          echo "$new_sha"
+          if [ "$initial_sha" == "$new_sha" ]; then
+            echo "No updates made"
+          else
+            echo "SHA changed, updating submodule"
```

Comment:
If this logic gets any more complicated, it should be extracted into a script that can be run locally by developers (for debugging and manual updates), then the workflow should run the script. Any committing and PR creation details should still go in the workflow file.